deploy/index.js

```diff
@@ -0,0 +1,374 @@
+        pass = false;
+      }
+    }
+    return pass;
+  });
+  return noPreOrBeta;
+};
+
+/**
+ * Sort release by date publish data
+ *
+ * @param  data - OctoKitResponse.data
+ * @returns data, or undefined
+ */
+const sortReleasesByDatePublishDate = (data) => {
+  data.sort((a, b) => a.published_at < b.published_at);
+  return data;
+};
+
+const getAssetLink = (assets, regex) => {
+  const jar = assets.filter((a) => regex.test(a.name));
+  if (jar.length > 0) {
+    let target = jar[0];
+    return { name: target.name, download_url: target.browser_download_url };
+  }
+};
+
+/**
+ * From API response, create a list of release names.
+ *
+ * @param  data - OctoKitResponse.data
+ * @returns data, or undefined
+ */
+const getReleaseData = (data) => {
+  // prefer tag_name to name, the latter might include spaces
+
+  const jarReg = /^flexy-cumulocity-connector-\d\.\d\.\d-\w+\.jar\b/g;
+  const configReg = /^CumulocityConnectorConfig.json/g;
+  const jvmRunReg = /^jvmrun/g;
+  const names = data.map((d) => {
+    return {
+      name: d.tag_name,
+      jar: getAssetLink(d.assets, jarReg),
+      configuration: getAssetLink(d.assets, configReg),
+      jvmRun: getAssetLink(d.assets, jvmRunReg),
+    };
+  });
+  return names;
+};
+
+/**
+ * Filter all release versions that do not include all asset files
+ *
+ *
+ * @param release - Github release data
+ *
+ * @returns True/False release includes all assets
+ */
+const FilterOnlyCompleteReleases = (release) => {
+  return release && release.jvmRun && release.jar && release.configuration;
+};
+
+/**
+ * Get the tag name of the latest release
+ *
+ * @returns tag name or undefined
+ */
+const getLatestReleaseTagName = async () => {
+  try {
+    let latest = await makeGithubApiLatestRequest();
+    if (latest && latest.status && latest.status === 200 && latest.data) {
+      if (latest.data.tag_name) {
+        return latest.data.tag_name;
+      }
+    }
+    console.err(
+      `Response for latest request unexpected ${JSON.stringify(
+        latest,
+        null,
+        2
+      )}. `
+    );
+  } catch (e) {
+    console.log(e.message);
+    return;
+  }
+};
+
+const transformDownloadUrlToS3Format = (asset) => {
+  const downloadUrlBase = S3_BASE_URL + asset.name + "/";
+  // one way to deep copy
+  let ret = JSON.parse(JSON.stringify(asset));
+  ret.jar.download_url = downloadUrlBase + asset.jar.name;
+  ret.configuration.download_url = downloadUrlBase + asset.configuration.name;
+  ret.jvmRun.download_url = downloadUrlBase + asset.jvmRun.name;
+  return ret;
+};
+
+/**
+ * Request release data and write manifest file.
+ *
+ * returns true for success or false for failure
+ */
+const writeReleaseManifestAndLatest = async () => {
+  let response;
+  try {
+    response = await makeGithubApi();
+  } catch (e) {
+    console.error("Unable to request Github API data.");
+    console.error(e.message);
+    process.exit(-1);
+  }
+  console.log(JSON.stringify(response, null, 3));
```

Comment:
magic number/null. What are null and 3 for?

Also, most JSON.stringify instances are using `2`. Why does this one use `3`?

Reply:
The last number is  number of white spaces for indentation of JSON strings. It's increased here for readability.

Reply:
A variable/constant should be introduced for these values to make the code more understandable.

Reply:
done